### PR TITLE
Executable tutorial

### PR DESCRIPTION
**Pull request by paraynaud**
- environnement @example 
- add required dependencies

### CODE REVIEW

File: docs/src/tutorial.md [hunk 1]
```diff
@@ -56,9 +59,12 @@ The chained structure that defines the neural network must be a subtype of `Knet
 
 ### Load datasets and define mini-batch
 In this example, we use the [MNIST](https://juliaml.github.io/MLDatasets.jl/stable/datasets/MNIST/) dataset from [MLDatasets.jl](https://github.com/JuliaML/MLDatasets.jl.git).
-```julia
+```@example KnetNLPModel
 using MLDatasets
 
+# download datasets without having to manually confirm it
```
Comment (dpo) [inline docs/src/tutorial.md:65]:
```suggestion
# download datasets without user intervention
```

**Review by dpo — CHANGES_REQUESTED** [review]

File: docs/src/tutorial.md [hunk 2]
```diff
@@ -22,13 +22,16 @@ KnetNLPModels is an interface between [Knet.jl](https://github.com/denizyuret/Kn
...
 using Knet
 
-struct Dense{T}
-  w :: Param{CuArray{Float32, 2, CUDA.Mem.DeviceBuffer}} # parameters of the layers
-  b :: Param{CuArray{Float32, 1, CUDA.Mem.DeviceBuffer}} # bias of the layer
+mutable struct Dense
+  w :: Param{Matrix{Float32}} # parameters of the layers
+  b :: Param{Vector{Float32}} # bias of the layer
+  # w :: Param{CuArray{Float32, 2, CUDA.Mem.DeviceBuffer}} # for GPU
+  # b :: Param{CuArray{Float32, 1, CUDA.Mem.DeviceBuffer}} # for GPU
```
[the remark this answers is not in the document]
Comment (dpo) [reply docs/src/tutorial.md:34]: Yes, for example.

**Comment by paraynaud** [comment]
I can't test it right now, all GPUs are used.

File: docs/src/tutorial.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (dpo) [reply docs/src/tutorial.md]: Does
```julia
struct Dense{T, Y}
  w :: Param{T}
  b :: Param{Y}
```
work?

[the remark this answers is not in the document]
Comment (paraynaud) [reply docs/src/tutorial.md]: It should work.
I typed it as accurate as I can.
I don't know about the performance (of Union), but I think it is more clear as it is than just `T` and `Y`.

**Comment by paraynaud** [comment]
I tested it on Atlas, and it worked.

File: README.md [hunk 3]
```diff
@@ -34,119 +38,8 @@ In addition, it provides tools to:
...
-reset_minibatch_train!(DenseNetNLPModel)
-```
-The size of the new minibatch is the size define earlier.
-
-The size of the training and test minibatch can be set to `1/p` the size of the dataset with:
-```julia
-set_size_minibatch!(DenseNetNLPModel, p) # p::Int > 1
-```
+## How to use
+Check the [tutorial](https://juliasmoothoptimizers.github.io/KnetNLPModels.jl/dev/tutorial/).
```
Comment (dpo) [inline README.md:42]: Better point to stable.

File: docs/Project.toml [hunk 4]
```diff
@@ -1,5 +1,10 @@
 [deps]
+CUDA = "052768ef-5323-5732-b1bb-66c8b64840ba"
```
Comment (dpo) [inline docs/Project.toml:2]: VMs don't always have a GPU. Why not run the docs on CPU?

File: docs/src/index.md [hunk 5]
```diff
@@ -18,3 +22,6 @@ In addition, it provides tools to:
 - switch the minibatch used to evaluate the neural network;
 - change the minibatch size;
 - measure the neural network's accuracy at the current `w`.
+
+## How to use
+Check the [tutorial](https://juliasmoothoptimizers.github.io/KnetNLPModels.jl/dev/tutorial/).
```
Comment (dpo) [inline docs/src/index.md:27]: Why does this point to dev?

File: docs/src/tutorial.md [hunk 6]
```diff
@@ -1,6 +1,7 @@
 # KnetNLPModels.jl Tutorial
 
 ## Synopsis
+KnetNLPModels is an interface between [Knet.jl](https://github.com/denizyuret/Knet.jl.git)'s classification neural networks and [NLPModels.jl](https://github.com/JuliaSmoothOptimizers/NLPModels.jl.git).
```
Comment (dpo) [inline docs/src/tutorial.md:4]: too much repetition

**Review by dpo — CHANGES_REQUESTED** [review]

File: docs/Project.toml (hunk 4 above)
Comment (paraynaud) [reply docs/Project.toml:2]: I added it to run the documentation (mainly the tutorial) on Atlas.

Comment (paraynaud) [reply docs/Project.toml]: (update) If we remove CUDA, we can't type the `Dense` structure like this:
```julia
struct Dense{
    T <: Union{Matrix{Float32}, CuArray{Float32, 2, CUDA.Mem.DeviceBuffer}}
    Y <: Union{Vector{Float32}, CuArray{Float32, 1, CUDA.Mem.DeviceBuffer}}
    }
  w :: Param{T}
  b :: Param{Y}
```
which makes documentation run fail.

Comment (dpo) [reply docs/Project.toml:2]: Use the more general typing I suggested above. I don't think the Union adds anything.

Comment (paraynaud) [reply docs/Project.toml:2]: It is done, I think we are ready to go.

**Review by dpo — APPROVED** [review]
Great, thank you!

**Pull request merged**